apps/prairielearn/src/lib/editors.js

```diff
@@ -95,119 +96,125 @@ export class Editor {
     callback(null);
   }
 
-  doEdit(callback) {
+  assertCanEdit() {
```

Comment:
Rather than duplicating all the conditions between `canEdit` and `assertCanEdit`, it's probably better to update the implementation of `canEdit` to use `assertCanEdit` internally. That is, something like this:

```ts
canEdit(callback) {
  try {
    assertCanEdit();
  } catch (err) {
    callback(err);
    return;
  }

  callback(null);
}

Reply:
`canEdit(callback)` is now using `assertCanEdit()` in [15e6a55](https://github.com/PrairieLearn/PrairieLearn/pull/9439/commits/15e6a551ace1b217ed76fec735c5306f682080d1).